srt-protocol/src/packet/control/srt.rs

```diff
@@ -208,18 +208,28 @@ impl SrtControlPacket {
             3 => Ok(KeyManagerRequest(SrtKeyMessage::parse(buf)?)),
             4 => Ok(KeyManagerResponse(SrtKeyMessage::parse(buf)?)),
             5 => {
-                // cut off null bytes at end
-                let mut string_bytes = vec![0; buf.remaining()];
-                buf.copy_to_slice(&mut string_bytes[..]);
-                let shrink_by = string_bytes
-                    .iter()
-                    .rev()
-                    .position(|a| *a != 0)
-                    .unwrap_or(string_bytes.len()); // all null
-
-                string_bytes.resize(string_bytes.len() - shrink_by, 0);
-
-                match String::from_utf8(string_bytes) {
+                // the stream id string is stored as 32-bit little endian words
+                // https://tools.ietf.org/html/draft-sharabayko-mops-srt-01#section-3.2.1.3
+                let mut bytes = Vec::with_capacity(buf.remaining());
+                let mut chunks = buf.chunk().chunks(4).peekable();
```

Comment:
This is probably fine, but is not technically correct--chunk() doesn't necessarily return all the bytes, as Buf doesn't have to be contiguous (it works because we're constructing it from a contiguous array). Perhaps a cleaner and more correct way to do this would be something along the lines:

```rust
if buf.remaining() % 4 != 0 { return Err(NotEnoughData) }

while buf.remaining > 4 {
     bytes.extend(buf.read_u32_le().to_be_bytes())
}
match buf.read_u32_le().to_be_bytes() {
    // your existing match logic here
}
```